Mathlib/Data/Sum/Basic.lean

```diff
@@ -464,6 +464,54 @@ theorem liftRel_swap_iff : LiftRel s r x.swap y.swap ↔ LiftRel r s x y :=
     exact h.swap, LiftRel.swap⟩
 #align sum.lift_rel_swap_iff Sum.liftRel_swap_iff
 
+theorem liftRel_def : LiftRel r s x y ↔ (∃ a, x = inl a ∧ ∃ c, y = inl c ∧ r a c) ∨
+(∃ b, x = inr b ∧ ∃ d, y = inr d ∧ s b d) := by
+refine' ⟨(fun H ↦ _), (fun H ↦ _)⟩
+· rcases H with (hac | hbd)
+  · exact Or.inl ⟨_, rfl, _, rfl, hac⟩
```

Comment:
This is very close to the lemma `LiftRel_iff` autogenerated by doing
`import Mathlib.Tactic.MkIffOfInductiveProp` and adding `@[mk_iff]` to `inductive LiftRel`. Have you considered using that either instead of this or to prove this more easily

Reply:
Oh I didn't know about that. That's a nice feature and really the kinda thing I was looking for. (I don't use this in the rest of the file really but I figured we should have it.)

Reply:
Indeed, its a feature I think is relatively not well known! So  I think there are many more places in the library it could be useful also but isnt yet

Reply:
I think a lot of the time you don't need it (as you see from the rest of the proofs) but yeah wow that should really be a gimmee. I find in Mathlib4 finding good tactic documentation is much harder.